lib/python/ray/worker.py

```diff
@@ -173,11 +191,21 @@ def get_object(self, objref):
     Args:
       objref (ray.ObjRef): The object reference of the value to retrieve.
     """
+
     if ray.lib.is_arrow(self.handle, objref):
-      result, segmentid = ray.lib.get_arrow(self.handle, objref)
+      ## this is the new codepath
+      buff, segmentid, metadata_offset = ray.libraylib.get_buffer(self.handle, objref)
+      metadata_size = np.frombuffer(buff, dtype="int64", count=1)[0]
+      metadata = np.frombuffer(buff, dtype="byte", offset=8, count=metadata_size)
+      data = np.frombuffer(buff, dtype="byte", offset=8+metadata_size)
+      serialized = libnumbuf.read_from_buffer(memoryview(data), bytearray(metadata), metadata_offset)
+      result = libnumbuf.deserialize_list(serialized)[0]
```

Comment:
for now, `assert len(libnumbuf.deserialize_list(serialized)) == 1` and document why it is a list (also in `put_object`)